apps/editing-toolkit/editing-toolkit-plugin/error-reporting/index.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * Error reporting from wp-admin / Gutenberg context for simple sites and Atomic.
+ *
+ * @package A8C\FSE
+ */
+
+namespace A8C\FSE\ErrorReporting;
+
+/**
+ * Inline  error handler that will capture errors before
+ * the main handler has a chance to. Errors are pushed
+ * to a global array called `_jsErr` which is then verified
+ * in the main handler. See `./index.js`.
+ */
+function head_error_handler() {
+	// The window.onerror handler can only catch events caused by the current origin, thus, it must be in the main document, or a script loaded from that origin
+	?><script type="text/javascript">
+	  window._headJsErrorHandler = function( errEvent ) {
+			window._jsErr = window._jsErr || [];
+			console.log(errEvent);
+			window._jsErr.push(errEvent);
+		}
+		window.addEventListener('error', window._headJsErrorHandler );
+
+		throw new Error('KABOOOOM');
+	</script>
+	<script crossorigin="anonymous" src="https://s0.wp.com/wp-content/plugins/corserror-head.js"></script>
+	<?php
+}
+add_action( 'admin_print_scripts', __NAMESPACE__ . '\head_error_handler' );
+
+function add_crossorigin_to_script_els( $tag ) {
+	if ( preg_match( '/<script\s.*src.*>/', $tag ) ) {
```

Comment:
I just realized we should exclude any third-party scripts like the ones loaded from Google etc (i.e `https://www.google.com/jsapi`).

Reply:
A better approach might be to restrict the `crossorigin` attribute to select wp.com subdomains (`s0`, `stats`, `widgets`).